src/main/java/de/dennisguse/opentracks/stats/TrackStatisticsUpdater.java

```diff
@@ -184,6 +198,12 @@ private void reset(TrackPoint trackPoint) {
         lastTrackPoint = null;
         altitudeBuffer.reset();
         speedBuffer.reset();
+        resetAverageHeartRate();
+    }
+
+    private void resetAverageHeartRate() {
+        averageHeartRateBPM = 0.0f;
+        totalHeartRateMillis = 0;
```

Comment:
Duration? Then it is explicit and makes reading the code easier.
Also it provides Duration.plus()

Reply:
Will do